x-pack/plugins/observability_solution/investigate_app/server/lib/collectors/fetcher.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { CollectorFetchContext } from '@kbn/usage-collection-plugin/server';
+import { StoredInvestigation } from '../../models/investigation';
+import { SO_INVESTIGATION_TYPE } from '../../saved_objects/investigation';
+import { computeMetrics } from './helpers/metrics';
+import { Usage } from './type';
+
+export const fetcher = async (context: CollectorFetchContext) => {
+  const finder = context.soClient.createPointInTimeFinder<StoredInvestigation>({
+    type: SO_INVESTIGATION_TYPE,
+    perPage: 1000,
+  });
```

Comment:
this could be potentially very memory heavy task, if we let's say store lot of data as part of investigation, i imagine storing images, screenshots, lens embeddables attributes JSON, LLM output, all of that data is stored in items right?

Reply:
That's possible indeed - I can reduce the batch size to something smaller, like 10?

Reply:
i think the status part could be fetched as an aggregation, the length of notes and items arrays is tricky to get since that isn't mapped.

Reply:
yeah let's reduce the perPage for now. i can't think of anything else. since we can't use `fields` as well. If we really want to optimise, we can save length of items and notes as separate fields and get those. that way items and notes we won't have to fetch. but i guess it might be overkill.

Reply:
addressed with https://github.com/elastic/kibana/pull/197659/commits/89bc3268c4c7389523bf24183ac1ec65ea458723